web/static.go

```diff
@@ -121,3 +139,27 @@ func unsupportedBrowserScriptHandler(w http.ResponseWriter, r *http.Request) {
 	templatesDir, _ := templates.GetTemplateDirectory()
 	http.ServeFile(w, r, filepath.Join(templatesDir, "unsupported_browser.js"))
 }
+
+func getOpenGraphMetaTags(c *Context, originalHTML string, titleTemplate string) string {
+	modifiedHTML := originalHTML
+	siteName := c.App.Srv().Config().TeamSettings.SiteName
+	siteDescription := model.TeamSettingsDefaultCustomDescriptionText
+	customSiteDescription := c.App.Srv().Config().TeamSettings.CustomDescriptionText
+	if customSiteDescription != nil && *customSiteDescription != "" {
+		siteDescription = *customSiteDescription
+	}
```

Comment:
The order of operations in this function is a bit jumbled. I went ahead and wrote what I'm thinking since it's difficult to describe:

```go
func getOpenGraphMetaTags(c *Context) string {
	siteName := model.TeamSettingsDefaultSiteName
	customSiteName := c.App.Srv().Config().TeamSettings.SiteName
	if customSiteName != nil && *customSiteName != "" {
		siteName = *customSiteName
	}

	siteDescription := model.TeamSettingsDefaultCustomDescriptionText
	customSiteDescription := c.App.Srv().Config().TeamSettings.CustomDescriptionText
	if customSiteDescription != nil && *customSiteDescription != "" {
		siteDescription = *customSiteDescription
	}

	titleTemplate := "<title>%s</title>"
	titleHTML := fmt.Sprintf(titleTemplate, html.EscapeString(siteName))

	descriptionTemplate := "<meta property=\"og:description\" content=\"%s\" />"
	descriptionHTML := fmt.Sprintf(ogDescriptionTemplate, html.EscapeString(siteDescription))

	return fmt.Sprintf("%s%s", titleHTML, descriptionHTML)
}
```

Then, I think we can unconditionally do the `bytes.ReplaceAll` in the caller function, since it should never equal the original title tag. Also, I think it's fine to duplicate `titleTemplate` among the two functions.

Reply:
Thank you for it, IMO, we don't need to add description meta tag if there is no specified right?

Let me know what you're thinking @mickmister

Reply:
> we don't need to add description meta tag if there is no specified right?

If a custom description is not provided, we fall back on the default description. Because of this, `siteDescription` is never an empty string. So we should unconditionally set the `og:description` tag with whichever description we're working with

Reply:
@mickmister Correct me if I'm wrong but current default description seems to be empty in here, 
https://github.com/mattermost/mattermost-server/blob/master/model/config.go#L118

Reply:
> @mickmister Correct me if I'm wrong but current default description seems to be empty in here, https://github.com/mattermost/mattermost-server/blob/master/model/config.go#L118

Indeed 🤔 It may be added later as well, which the code takes into account. Current code looks good :+1: